packages/server/graphql/intranetSchema/mutations/hardDeleteUser.ts

```diff
@@ -0,0 +1,266 @@
+import {GraphQLID, GraphQLNonNull, GraphQLString} from 'graphql'
+import {GQLContext} from '../../graphql'
+import {requireSU} from '../../../utils/authorization'
+import getRethink from '../../../database/rethinkDriver'
+import getPg from '../../../postgres/getPg'
+import softDeleteUser from '../../mutations/helpers/softDeleteUser'
+import DeleteUserPayload from '../../types/DeleteUserPayload'
+import {getUserById} from '../../../postgres/queries/getUsersByIds'
+import {getUserByEmail} from '../../../postgres/queries/getUsersByEmails'
+import blacklistJWT from '../../../utils/blacklistJWT'
+import {toEpochSeconds} from '../../../utils/epochTime'
+import TeamMemberId from 'parabol-client/shared/gqlIds/TeamMemberId'
+
+const generateDeletedUserId = () => `deletedUser-${Date.now()}`
+
+const hardDeleteUser = {
+  type: GraphQLNonNull(DeleteUserPayload),
+  description: 'hard deletes a user and all its associated objects',
+  args: {
+    userId: {
+      type: GraphQLID,
+      description: 'a userId'
+    },
+    email: {
+      type: GraphQLID,
+      description: 'the user email'
+    },
+    reasonText: {
```

Comment:
i think we can get rid of reasonText since the User object will get deleted unless i missed a place where it is stored?

Reply:
the `reasonText` is passed to `softDeleteUser` resolver which passes it to `segment.track` method